### PR TITLE
Add llc agreement

### PR DESCRIPTION
**Pull request by guibescos**
Adds an intruction join_dao_lcc where the user needs to sign a hash of the DAO LLC agreement.
This agreement needs to be signed before the user is able to vote or stake.

### CODE REVIEW

File: staking/programs/staking/src/lib.rs [hunk 1]
```diff
@@ -123,6 +124,7 @@ pub mod staking {
 
         stake_account_metadata.lock = lock;
         stake_account_metadata.transfer_epoch = None;
+        stake_account_metadata.is_llc_member = false;
```
Comment (jayantk) [inline staking/programs/staking/src/lib.rs:127]: i suggest saving the hash of the agreement they agreed to. That would e.g., let governance change the agreement text and enforce that user re-sign the new agreement. Could be useful at some point in the future.

File: staking/app/StakeConnection.ts [hunk 2]
```diff
@@ -708,6 +734,10 @@ export class StakeConnection {
       );
     }
 
+    if (!stakeAccount || !this.isLlcMember(stakeAccount)) {
```
Comment (jayantk) [inline staking/app/StakeConnection.ts:737]: seems like there's a block of code above this that's duplicated a few times. consider extracting out a helper method.

**Review by jayantk — APPROVED** [review]
nice

File: staking/app/StakeConnection.ts (hunk 2 above)
Comment (guibescos) [reply staking/app/StakeConnection.ts:740]: Future PR

**Pull request merged**